From 60b58c678acd14816d9aa0a0d0099f8e132943fe Mon Sep 17 00:00:00 2001
From: Jes Sorensen <Jes.Sorensen@redhat.com>
Date: Mon, 1 Sep 2014 09:25:36 -0400
Subject: [block] submit_bio_wait() conversions

Message-id: <1409563584-9000-35-git-send-email-Jes.Sorensen@redhat.com>
Patchwork-id: 89971
O-Subject: [RHEL7.1 PATCH 34/82] block: submit_bio_wait() conversions
Bugzilla: 1085530
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

From: Kent Overstreet <kmo@daterainc.com>

It was being open coded in a few places.

Signed-off-by: Kent Overstreet <kmo@daterainc.com>
Cc: Jens Axboe <axboe@kernel.dk>
Cc: Joern Engel <joern@logfs.org>
Cc: Prasad Joshi <prasadjoshi.linux@gmail.com>
Cc: Neil Brown <neilb@suse.de>
Cc: Chris Mason <chris.mason@fusionio.com>
Acked-by: NeilBrown <neilb@suse.de>
(partial backport of 33879d4512c021ae65be9706608dacb36b4687b1)
Signed-off-by: Jes Sorensen <Jes.Sorensen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/md/md.c b/drivers/md/md.c
index 5f1a5ce..e396c27 100644
--- a/drivers/md/md.c
+++ b/drivers/md/md.c
@@ -776,20 +776,12 @@ void md_super_wait(struct mddev *mddev)
  finish_wait(&mddev->sb_wait, &wq);
 }
 
-static void bi_complete(struct bio *bio, int error)
-{
- complete((struct completion*)bio->bi_private);
-}
-
 int sync_page_io(struct md_rdev *rdev, sector_t sector, int size,
    struct page *page, int rw, bool metadata_op)
 {
  struct bio *bio = bio_alloc_mddev(GFP_NOIO, 1, rdev->mddev);
- struct completion event;
  int ret;
 
- rw |= REQ_SYNC;
-
  bio->bi_bdev = (metadata_op && rdev->meta_bdev) ?
   rdev->meta_bdev : rdev->bdev;
  if (metadata_op)
@@ -801,11 +793,7 @@ int sync_page_io(struct md_rdev *rdev, sector_t sector, int size,
  else
   bio->bi_sector = sector + rdev->data_offset;
  bio_add_page(bio, page, size, 0);
- init_completion(&event);
- bio->bi_private = &event;
- bio->bi_end_io = bi_complete;
- submit_bio(rw, bio);
- wait_for_completion(&event);
+ submit_bio_wait(rw, bio);
 
  ret = test_bit(BIO_UPTODATE, &bio->bi_flags);
  bio_put(bio);
-- 
1.7.1